xchange-binance/src/main/java/org/knowm/xchange/binance/service/BinanceTradeService.java

```diff
@@ -321,12 +332,39 @@ public OpenOrdersParams createOpenOrdersParams() {
               "You need to provide the currency pair and the order id to query an order.");
         }
 
-        orders.add(
-            BinanceAdapters.adaptOrder(
-                super.orderStatus(
-                    orderQueryParamCurrencyPair.getCurrencyPair(),
-                    BinanceAdapters.id(orderQueryParamCurrencyPair.getOrderId()),
-                    null)));
+        Order order = BinanceAdapters.adaptOrder(
+                        super.orderStatus(
+                          orderQueryParamCurrencyPair.getCurrencyPair(),
+                          BinanceAdapters.id(orderQueryParamCurrencyPair.getOrderId()),
+                          null
+                        )
+                      );
+
+        // Calculate fees by collecting trades associated with this order
+        BinanceTradeHistoryParams tradeHistParams = new BinanceTradeHistoryParams(orderQueryParamCurrencyPair.getCurrencyPair());
+        UserTrades tradeHist = getTradeHistory(tradeHistParams);
```

Comment:
scanning the entire trade history for each order will not we acceptable for performance reasons